presto-tests/src/main/java/io/prestosql/tests/AbstractTestJoinQueries.java

```diff
@@ -2192,23 +2192,28 @@ public void testSemiJoinPredicateMoveAround()
                 "    orderkey % 2 = 0");
     }
 
-    @Test
+    @Test(timeOut = 120_000)
     public void testInnerJoinWithEmptyBuildSide()
     {
-        MaterializedResult actual = computeActual(
+        // TODO: increase lineitem schema size when build side short-circuit is fixed
+        assertQuery(
                 noJoinReordering(),
-                "WITH small_part AS (SELECT * FROM part WHERE name = 'a') " +
-                        "SELECT lineitem.orderkey FROM lineitem INNER JOIN small_part ON lineitem.partkey = small_part.partkey");
-
-        assertEquals(actual.getRowCount(), 0);
+                "WITH empty_table AS (SELECT partkey as value FROM part WHERE name = 'a'), " +
+                        "slow_table AS (SELECT partkey as value FROM tpch.\"sf0.1\".lineitem) " +
+                        "SELECT slow_table.value FROM slow_table INNER JOIN empty_table ON slow_table.value = empty_table.value",
+                "SELECT 0 WHERE false");
     }
 
-    @Test
+    @Test(timeOut = 120_000)
     public void testRightJoinWithEmptyBuildSide()
     {
+        // TODO: increase lineitem schema size when build side short-circuit is fixed
```

Comment:
change this to `// TODO use blackhole slow table and reduce test timeout when build side short-circuit is fixed`

Reply:
IMO using blackhole connector here would make test more verbose and longer.